runtime/lua/vim/_editor.lua

```diff
@@ -714,7 +720,119 @@ function vim._on_key(buf, typed_buf)
   end
 end
 
---- Generates a list of possible completions for the string.
+--- Convert UTF-32, UTF-16 or UTF-8 {index} to byte index.
+--- If {no_out_of_range} is true
+--- then then an out of range index will return byte length
+--- instead of throwing an error.
+---
+--- Invalid UTF-8 and NUL is treated like in |vim.str_utfindex()|.
+--- An {index} in the middle of a UTF-16 sequence is rounded upwards to
+--- the end of that sequence.
+---@param s string
+---@param encoding "utf-8"|"utf-16"|"utf-32"
+---@param index integer
+---@param no_out_of_range? boolean
```

Comment:
I'm not sure use of positional arguments for this kind of boolean flag or options is the best choice. Can we use `opt` table for this?

Reply:
These types of functions are typically used in hot loops where the table allocation may be significant.